lib/jekyll/converters/scss.rb

```diff
@@ -75,11 +75,28 @@ def sass_dir_relative_to_site_source
       end
 
       def sass_load_paths
+        paths = user_sass_load_paths + [sass_dir_relative_to_site_source]
+
         if safe?
-          [sass_dir_relative_to_site_source]
-        else
-          (user_sass_load_paths + [sass_dir_relative_to_site_source]).uniq
-        end.select { |load_path| File.directory?(load_path) }
+          # Sanitize paths to prevent any attack vectors (.e.g. `/**/*`)
+          paths.map! { |path| Jekyll.sanitized_path(@config["source"], path) }
```

Comment:
What would happen if a malicious user set `source: ../../../../../password`? Aren't we using untrusted user input here to sanitize user input? (Or is that checked further down the stack @parkr?)